CMakeLists.txt

```diff
@@ -27,3 +31,18 @@ add_subdirectory(src)
 add_subdirectory(doc)
 add_subdirectory(test)
 add_subdirectory(cpack)
+
+if(SYSTEM_NAME STREQUAL "FreeBSD")
+  # use a linker from ports, e.g. lld50 from llvm
+  message("use a linker built from ports")
+
+  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") # using Clang
+    # Ensure that FreeBSD build uses current clang from ports
+    # which is required to correctly link the required libraries.
```

Comment:
Can you provide more info on why this is needed? A link to a clang issue or something?

Reply:
It has been some time since I have issued this pull request, but if I remember correctly, the version of clang bundled in FreeBSD 11.1 base (version 4.0) is built without knowledge of libcurl, why the "-lcurl" flag required at cryfs build cannot be found directly. Using a compiler built from ports tree which includes the currently built libs by default resolves this pitfall, though here are other ways to include curl.

Reply:
I'm a FreeBSD newbie. Can you elaborate on this please? Which clang port? There are several. I also don't see anything in CMakeLists.txt that discusses freebsd 11.1. Maybe I'm on the wrong branch.